modin/core/dataframe/algebra/groupby.py

```diff
@@ -373,16 +412,196 @@ def try_filter_dict(agg_func, df):
+                else:
+                    agg_key = [col]
+
+                result = func(grp_obj[agg_key])
+                result_has_id_level = result.columns.names[0] == cls.ID_LEVEL_NAME
+                insert_id_levels |= result_has_id_level
+
+                if col_renaming_required:
+                    func_name = str(func) if func_name is None else func_name
+                    if result_has_id_level:
+                        result.columns = pandas.MultiIndex.from_tuples(
+                            [
+                                # `old_col[0]` stores values from the 'id'
+                                # level, the ones we want to preserve here
+                                (old_col[0], col, func_name)
+                                for old_col in result.columns
+                            ],
+                            names=[
+                                result.columns.names[0],
+                                result.columns.names[1],
+                                None,
+                            ],
+                        )
+                    else:
+                        result.columns = pandas.MultiIndex.from_tuples(
+                            [(col, func_name)] * len(result.columns),
+                            names=[result.columns.names[0], None],
+                        )
+
+                external_results.append(result)
+
+            if insert_id_levels:
+                # As long as any `result` has an id-level we have to insert the level
+                # into every `result` so the number of levels matches
+                for idx in range(len(external_results)):
+                    if external_results[idx].columns.names[0] != cls.ID_LEVEL_NAME:
+                        external_results[idx] = pandas.concat(
+                            [external_results[idx]],
+                            keys=[cls.ID_LEVEL_NAME],
+                            names=[cls.ID_LEVEL_NAME],
+                            axis=1,
+                            copy=False,
+                        )
+
+                if native_agg_res is not None:
+                    native_agg_res = pandas.concat(
+                        [native_agg_res],
+                        keys=[cls.ID_LEVEL_NAME],
+                        names=[cls.ID_LEVEL_NAME],
+                        axis=1,
+                        copy=False,
+                    )
+
+            native_res_part = [] if native_agg_res is None else [native_agg_res]
+            result = pandas.concat(
+                [*native_res_part, *external_results], axis=1, copy=False
+            )
+
+            # The order is naturally preserved if there's no external aggregations
+            if preserve_aggregation_order and len(external_aggs):
+                result = result.reindex(result_columns, axis=1)
+            return result
+
+        return aggregate_on_dict
+
+    @classmethod
+    def is_external_aggregation(cls, func):
```

Comment:
that naming seems confusing to me, why a registered function is considered `external`?.. external to what?